src/requests/form_builder.rs

```diff
@@ -50,13 +53,31 @@ impl FormBuilder {
         }
     }
 
+    fn add_file_from_memory<'a, N>(
+        self,
+        name: N,
+        file_name: String,
+        data: Vec<u8>,
+    ) -> Self
+    where
+        N: Into<Cow<'a, str>>,
+    {
+        Self {
+            form: self.form.part(
+                name.into().into_owned(),
+                file_from_memory_to_part(data, file_name),
+            ),
+        }
+    }
+
     pub fn build(self) -> Form {
         self.form
     }
 }
 
 pub(crate) enum FormValue {
     File(PathBuf),
+    Memory { file_name: String, data: Vec<u8> },
```

Comment:
Files can be too large to clone their content each time, so I think that it's better to use `&'a [u8]` instead of `Vec<u8>`, right?

Reply:
Best i could do was make `Cow<'static, [u8]>` because request's `Part` requirement

Reply:
Let me check today, I think that it's possible to make `&'a [u8]`.

Reply:
[I've asked](https://github.com/seanmonstar/reqwest/issues/861) about it, so you're right, reqwest doesn't accept arbitrary lifetimes. It could be even better to accept streams, but for now, unfortunately, I think the approach with `Cow<'static, u8>` is the best.